### PR TITLE
AABB_tree: make accelerate distance queries default

### LINKED ISSUE #4008: Lazy AABB::accelerate_distance_queries()
## Issue Details

It is too easy to forget to call `AABB::accelerate_distance_queries()` before a distance query.

AABB_tree should:
  - either call it lazily the first time a distance query is made,
  - or display a warning message at runtime, about the developer forgetting to call it.

## TODO
- [x] add a function AABB::do_not_accelerate_distance_queries(), that stores a Boolean member variable,
- [x] call AABB::accelerate_distance_queries() automatically (lazily) if the Boolean is not set.
- [x] announce in the release note a change of behavior (breaking change) as the default will be with accelerate distance queries.
## Environment
* CGAL version: all versions

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Make the `accelerate_distance_queries()` lazy and add a function to disable it.

## Release Management

* Affected package(s):AABB_tree
* Issue(s) solved (if any): fix #4008
* Feature/Small Feature (if any):
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership:

### CODE REVIEW

File: Surface_mesh_segmentation/include/CGAL/internal/Surface_mesh_segmentation/SDF_calculation.h [hunk 1]
```diff
@@ -173,9 +173,6 @@ class SDF_calculation
     }
     tree.build();
 
-    if(build_kd_tree) {
```
Comment (sloriot) [inline Surface_mesh_segmentation/include/CGAL/internal/Surface_mesh_segmentation/SDF_calculation.h]: Looks suspicious

File: AABB_tree/include/CGAL/AABB_tree.h [hunk 2]
```diff
@@ -185,7 +185,7 @@ namespace CGAL {
       clear_nodes();
 			m_primitives.clear();
 			clear_search_tree();
-			m_default_search_tree_constructed = false;
+			m_default_search_tree_constructed = m_use_lazy_kd_tree;
```
Comment (sloriot) [inline AABB_tree/include/CGAL/AABB_tree.h]: This should be `true` i.e. restore the default behavior.

File: AABB_tree/include/CGAL/AABB_tree.h [hunk 3]
```diff
@@ -439,6 +439,9 @@ namespace CGAL {
 		/// a point set taken on the internal primitives
 		/// returns `true` iff successful memory allocation
 		bool accelerate_distance_queries() const;
+		///Turns off the lazy construction of the internal search tree.
+		/// Must be called before the first distance query.
```
Comment (sloriot) [inline AABB_tree/include/CGAL/AABB_tree.h]: Since the tree is cleared, it is not a mandatory condition.

File: AABB_tree/include/CGAL/AABB_tree.h [hunk 4]
```diff
@@ -590,6 +593,7 @@ namespace CGAL {
 		// search KD-tree
 		mutable const Search_tree* m_p_search_tree;
 		mutable bool m_search_tree_constructed;
+		mutable bool m_use_lazy_kd_tree;
```
Comment (sloriot) [inline AABB_tree/include/CGAL/AABB_tree.h]: it is not clear why you need another variable.

File: Surface_mesh_segmentation/include/CGAL/internal/Surface_mesh_segmentation/SDF_calculation.h (hunk 1 above)
Comment (maxGimeno) [reply Surface_mesh_segmentation/include/CGAL/internal/Surface_mesh_segmentation/SDF_calculation.h]: How so ?

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-18.shtml

File: AABB_tree/include/CGAL/AABB_tree.h [hunk 5]
```diff
@@ -623,7 +623,7 @@ namespace CGAL {
                            T&& ... t)
 		: m_traits()
     , m_primitives()
-                , m_p_root_node(nullptr)
+		, m_p_root_node(nullptr)
 		, m_p_search_tree(nullptr)
```
Comment (lrineau) [inline AABB_tree/include/CGAL/AABB_tree.h]: I still see an indentation issue, line 625,

File: AABB_tree/include/CGAL/AABB_tree.h [hunk 6]
```diff
@@ -623,7 +623,7 @@ namespace CGAL {
...
 		, m_search_tree_constructed(false)
     , m_default_search_tree_constructed(true)
```
Comment (lrineau) [inline AABB_tree/include/CGAL/AABB_tree.h]: Line 629 is also indented strange. It seems this file has a lot of indentation issues

Comment (sloriot) [reply AABB_tree/include/CGAL/AABB_tree.h]: That's probably tabs vs spaces.

Comment (maxGimeno) [reply AABB_tree/include/CGAL/AABB_tree.h]: It is. My IDE replaces tabs by spaces, leading to that kind of indentation changes everytime I touch an old file.

Comment (MaelRL) [reply AABB_tree/include/CGAL/AABB_tree.h]: This particular change seems to be the opposite, though: spaces are being replaced with tabs. Would be cool to have #3895 resurrected.

Comment (sloriot) [reply AABB_tree/include/CGAL/AABB_tree.h]: I'd be happy to get it back in shape if this time I'm sure it will be merged.

**Pull request merged**